### ISSUE TITLE
multi_p(@promises) issues commands in wrong order

### ISSUE THREAD

**Issue by Yenya** [jhthorsen/mojo-redis#68]
When I try to issue commands in a MULTI/EXEC transaction according to the example at https://metacpan.org/pod/Mojo::Redis::Database#multi_p, it seems that `multi_p(@promises)` sends the in-transaction commands to Redis first, and only then issues a MULTI  command. Which means the commands are executed outside the transaction. The test script follows:

```
#!env perl

use Mojo::Base -signatures;
use Mojo::Redis;

my $redis = Mojo::Redis->new(shift)
        or die "Can't connect to Redis";

my $db = $redis->db;

# $db->multi_p->then(sub { # this works, but the following does not:
$db->multi_p(
        $db->set_p('testkey' => 42, EX => 100),
)->then(sub {
        $db->exec_p;
})->then(sub {
        say "exec returned " . @_; # just the # of arguments
})->wait;
```

When I run `MOJO_REDIS_DEBUG=1 ./txn-test.pl my_database_connect_string`, I get the following output:

```
[1a8d36dd5c2d5a426c7cbd3a7ea29155] CONNECTING redis://... (blocking=0)
[1a8d36dd5c2d5a426c7cbd3a7ea29155] <<< (*2\r\n$4\r\nAUTH\r\n...\r\n)
[1a8d36dd5c2d5a426c7cbd3a7ea29155] <<< (*3\r\n$3\r\nSET\r\n$7\r\ntestkey\r\n$2\r\n42\r\n)
[1a8d36dd5c2d5a426c7cbd3a7ea29155] <<< (*1\r\n$5\r\nMULTI\r\n)
[1a8d36dd5c2d5a426c7cbd3a7ea29155] >>> (+OK\r\n+OK\r\n+OK\r\n)
[1a8d36dd5c2d5a426c7cbd3a7ea29155] <<< (*1\r\n$4\r\nEXEC\r\n)
[1a8d36dd5c2d5a426c7cbd3a7ea29155] >>> (*0\r\n)
exec returned 1
[0] CLOSED
```

Using the syntax `$db->multi_p->then(sub { $db->set_p(...); })` instead of `$db->multi_p($db->set_p(...))` works as expected - the MULTI command is issued first, and only then the SET command. I think the `multi_p(@promises)` syntax should be removed, because at the time `@promises` are passed to `multi_p()`, they can already be in flight and sent to Redis.

**Comment by Grinnz** [comment]
This seems accurate. If there is any convenience syntax it should probably be with callbacks so they are not executed until the MULTI command is sent. But since promise chaining can already achieve this, I see no need for the convenience.

**Comment by jhthorsen** [comment]
This is fixed in #70. Thank you 👍
